docs/okeps/okep-5224-connecting-udns/okep-5224-connecting-udns.md

```diff
@@ -0,0 +1,1652 @@
+  // if you plan to deploy 100 nodes, set the layer3NetworkPrefix to /24 (200+ IPs)
+  // if you plan to deploy 1000 nodes, set the layer3NetworkPrefix to /21 (2000+ IPs)
+  // if you plan to deploy 5000 nodes, set the layer3NetworkPrefix to /18 (10000+ IPs)
+  // This field restricts the maximum number of nodes that can be deployed in the cluster
+  // and hence its good to plan this value carefully along with the CIDR.
+  //
+  // +kubebuilder:validation:Minimum=1
+  // +kubebuilder:validation:Maximum=127
+  // +required
+  Layer3NetworkPrefix int32 `json:"layer3NetworkPrefix"`
+}
+
+// ConnectivityType represents the different connectivity types that can be enabled for connected networks.
+// +kubebuilder:validation:Enum=PodNetwork;ClusterIPServiceNetwork
+type ConnectivityType string
+
+const (
+    // PodNetwork enables direct pod-to-pod communication across connected networks.
+    PodNetwork ConnectivityType = "PodNetwork"
+
+    // ClusterIPServiceNetwork enables ClusterIP service access across connected networks.
+    ClusterIPServiceNetwork ConnectivityType = "ClusterIPServiceNetwork"
+)
+
+// StatusType represents the status of a ClusterNetworkConnect.
+// +kubebuilder:validation:Enum=Success;Failure
+type StatusType string
+
+const (
+    // Success indicates that the ClusterNetworkConnect has been successfully applied.
+    Success StatusType = "Success"
+
+    // Failure indicates that the ClusterNetworkConnect has failed to be applied.
+    Failure StatusType = "Failure"
+)
+
+// ClusterNetworkConnectStatus defines the observed state of ClusterNetworkConnect.
+type ClusterNetworkConnectStatus struct {
+    // status is a concise indication of whether the ClusterNetworkConnect
+    // resource is applied with success.
+    // +kubebuilder:validation:Optional
+    Status StatusType `json:"status,omitempty"`
+
+    // conditions is an array of condition objects indicating details about
+    // status of ClusterNetworkConnect object.
+    // +patchMergeKey=type
+    // +patchStrategy=merge
+    // +listType=map
+    // +listMapKey=type
+    Conditions []metav1.Condition `json:"conditions,omitempty"`
+}
+
+// ClusterNetworkConnectList contains a list of ClusterNetworkConnect.
+// +kubebuilder:object:root=true
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+type ClusterNetworkConnectList struct {
+    metav1.TypeMeta `json:",inline"`
+    metav1.ListMeta `json:"metadata,omitempty"`
+    Items           []ClusterNetworkConnect `json:"items"`
+}
+```
+
+#### Validation Field Checks
+
+In the implementation, after parsing the `ClusterNetworkConnect`
+object, we must ensure the following:
+
+* `connectSubnets` must not conflict with:
+  1) pod subnets of the selected networks to be connected
+  2) transit subnets of the selected networks to be connected
+  3) service CIDR range used in the cluster
+  4) join subnet of the selected networks to be connected
+  5) masquerade subnet range used in the cluster
```

Comment:
missing node subnet

Reply:
PTAL https://github.com/ovn-kubernetes/ovn-kubernetes/pull/5246#discussion_r2427339596 I'm a bit confused with if you mean the node-subnet annotation which is the pod subnet OR node IPs machine pool config?

Reply:
Kubernets node IPs/node underlay subnet